migrator/src/main/java/com/redhat/lightblue/migrator/Main.java

```diff
@@ -1,81 +1,111 @@
-            }
-            checker.setConsistencyCheckerName(System.getProperty("name"));
-            checker.setHostName(System.getProperty("hostname"));
-            checker.setConfigPath(System.getProperty("config"));
-            checker.setMigrationConfigurationEntityVersion(System.getProperty("configversion"));
-            checker.setMigrationJobEntityVersion(System.getProperty("jobversion"));
-            checker.setSourceConfigPath(System.getProperty("sourceconfig"));
-            checker.setDestinationConfigPath(System.getProperty("destinationconfig"));
-        } catch (ParseException e) {
-            HelpFormatter formatter = new HelpFormatter();
-            formatter.printHelp(ConsistencyCheckerCLI.class.getSimpleName(), options, true);
-            System.out.println("\n");
-            System.out.println(e.getMessage());
-            System.exit(1);
-        }
-
-        return checker;
-    }
-}
+package com.redhat.lightblue.migrator;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.log4j.PropertyConfigurator;
+
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.OptionBuilder;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.PosixParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.HelpFormatter;
+
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.LightblueClient;
+import com.redhat.lightblue.client.http.LightblueHttpClient;
+import com.redhat.lightblue.client.hystrix.LightblueHystrixClient;
+
+import static com.redhat.lightblue.client.expression.query.ValueQuery.withValue;
+import static com.redhat.lightblue.client.projection.FieldProjection.includeFieldRecursively;
+
+public class Main {
+
+    private static final Logger LOGGER=LoggerFactory.getLogger(Main.class);
+
+    public static void main(String[] args) throws Exception {
+        processArguments(args);
+
+        ConsistencyChecker checker = buildConsistencyChecker();
+        checker.run();
+    }
+
+    public static ConsistencyChecker buildConsistencyChecker() {
+        ConsistencyChecker checker = new ConsistencyChecker();
+        checker.setConsistencyCheckerName(System.getProperty("name"));
+        checker.setHostName(System.getProperty("hostname"));
+        checker.setConfigPath(System.getProperty("config"));
+        checker.setMigrationConfigurationEntityVersion(System.getProperty("configversion"));
+        checker.setMigrationJobEntityVersion(System.getProperty("jobversion"));
+        checker.setSourceConfigPath(System.getProperty("sourceconfig"));
+        checker.setDestinationConfigPath(System.getProperty("destinationconfig"));
+        return checker;
```

Comment:
Oh, I see it. I don't think I originated it, but I do see where you just kept the existing code. Puppet is passing in those parameters as jvm arguments I think. You can disregard this comment.